cli/src/main/java/org/adamalang/cli/devbox/DevBoxAdama.java

```diff
@@ -271,10 +274,15 @@ public void onSetupComplete(CoreStream stream) {
           io.error("adama|connection; It took over " + Math.round((delta / 100.0)) / 10.0 + " seconds to establish a connection");
         }
         PERF_LOG.error(entry.toString());
+        RxPubSub.instance.subscribe(rxResponderId, responder);
       }
 
       @Override
       public void status(StreamStatus status) {
+        if (status == StreamStatus.Disconnected) {
```

Comment:
I am not sure if there is a better trigger to remove the data responder or if this at all what you had in mind. I don't see a lot of singletons, but without passing classes all around I didn't see a good way to make the threads talk to eachother.

Reply:
In terms of singleton, you just have to update the plumbing and pass a shared object between the threads.

For the signalling, you want to mirror the unsubscribe to the CoreStream's close. So, when got.close() happens, that is when to end the subscription. In this case, you also have to deal with the event that the CoreStream died, so you should also unsubscribe on the failure.